src/vs/editor/browser/viewParts/indentGuides/indentGuides.ts

```diff
@@ -68,14 +68,10 @@ export class IndentGuidesOverlay extends DynamicViewOverlay {
 		return true;
 	}
 	public override onCursorStateChanged(e: viewEvents.ViewCursorStateChangedEvent): boolean {
-		const selection = e.selections[0];
-		const newPosition = selection.getPosition();
-		if (!this._primaryPosition?.equals(newPosition)) {
-			this._primaryPosition = newPosition;
-			return true;
-		}
+		const newPositions = e.selections.map(s => s.getPosition());
+		this._cursorPositions = newPositions;
 
-		return false;
+		return !this._cursorPositions[0]?.equals(newPositions[0]);
```

Comment:
If think you should rather check if the list of previous cursors equals the list of new cursors.

Reply:
My idea was to keep as much of the original behavior as possible where changes are not required. If I check the whole list that would mean the method will return true when some of the none primary cursors are changed. From what I saw in the code this is not necessary, but if my assumption is not correct I will be happy to change it.